kite-tools-parent/kite-tools/src/main/java/org/kitesdk/cli/commands/TarImportCommand.java

```diff
@@ -0,0 +1,167 @@
+import org.kitesdk.data.Dataset;
+import org.kitesdk.data.DatasetDescriptor;
+import org.kitesdk.data.DatasetWriter;
+import org.kitesdk.data.spi.DatasetRepository;
+import org.slf4j.Logger;
+
+import java.io.IOException;
+import java.util.List;
+
+@Parameters(commandDescription="Import files in tarball into a Dataset")
+public class TarImportCommand extends BaseDatasetCommand {
+
+  private static final List<String> SUPPORTED_TAR_COMPRESSION_TYPES = Lists.newArrayList("none", "gzip", "bzip2");
+
+  public TarImportCommand(Logger console) {
+    super(console);
+  }
+
+  @Parameter(description="<tar path> <dataset name>")
+  List<String> targets;
+
+  @Parameter(names="--compression", description="Compression type (none, gzip, bzip2)")
+  String compressionType = "none";
+
+  @Override
+  public int run() throws IOException {
+    Preconditions.checkArgument(targets != null && targets.size() == 2,
+        "Tar path and target dataset name are required.");
+
+    Path source = qualifiedPath(targets.get(0));
+    FileSystem sourceFS = source.getFileSystem(getConf());
+    Preconditions.checkArgument(sourceFS.exists(source),
+        "Tar path does not exist: " + source);
+
+    Preconditions.checkArgument(SUPPORTED_TAR_COMPRESSION_TYPES.contains(compressionType),
+        "Compression type " + compressionType + " is not supported");
+
+    String datasetName = targets.get(1);
+
+    int success = 0;
+
+    DatasetRepository repo = getDatasetRepository();
+
+    Dataset<GenericRecord> tarImport = null;
+    if (!repo.exists(namespace, datasetName)) {
+      console.info("Creating dataset {} in repo {}", datasetName, repo.getUri());
+      tarImport = repo.create(namespace, datasetName,
+          new DatasetDescriptor.Builder()
+              .schema(new Schema.Parser().parse(
+                  "{\n" +
+                      "  \"type\": \"record\",\n" +
+                      "  \"name\": \"tarimport\",\n" +
+                      "  \"fields\": [\n" +
+                      "  {\n" +
+                      "    \"type\": \"string\",\n" +
+                      "    \"name\": \"filename\"\n" +
+                      "  },\n" +
+                      "  {\n" +
+                      "    \"type\": \"bytes\",\n" +
+                      "    \"name\": \"filecontent\"\n" +
+                      "  }\n" +
+                      "  ]\n" +
+                      "}"
+              ))
+              .build()
+      );
+    } else {
+      console.info("Using existing dataset {} at {}", datasetName, repo.getUri());
+      tarImport = repo.load(namespace, datasetName);
+    }
+
+    DatasetWriter<GenericRecord> writer = tarImport.newWriter();
+
+    // Create a Tar input stream wrapped in appropriate decompressor
+    // Enhancement would be to use native compression libs
+    TarInputStream tis = null;
+    if (compressionType.equals("gzip")) {
+      tis = new TarInputStream(new GzipCompressorInputStream(sourceFS.open(source)));
```

Comment:
Instead of soruceFS.open, we now have an [`open` utility method](https://github.com/kite-sdk/kite/blob/master/kite-tools-parent/kite-tools/src/main/java/org/kitesdk/cli/commands/BaseCommand.java#L177) as well as a [`output` utility method](https://github.com/kite-sdk/kite/blob/master/kite-tools-parent/kite-tools/src/main/java/org/kitesdk/cli/commands/BaseCommand.java#L76) that will do the right thing with less logic in the command.